test/functional/pdo_sqlsrv/pdo_AE_functions.inc

```diff
@@ -0,0 +1,532 @@
+                $AEstmt->bindParam(1, $thresholds[$type], $PDOType);
+                $nonAEstmt = $conn->prepare($nonAEQuery);
+                $nonAEstmt->bindParam(1, $thresholds[$type], $PDOType);
+            } catch (PDOException $error) {
+                print_r($error);
+                die("Preparing/binding statements for comparison failed");
+            }
+
+            compareResults($AEstmt, $nonAEstmt, $key, $encryptionType, $attestation, $comparison);
+        }
+    }
+}
+
+// testPatternMatch selects based on a pattern in the WHERE clause and compares
+// the results between encrypted and non-encrypted columns, checking that the
+// results are identical
+function TestPatternMatch($conn, $tableName, $patterns, $dataTypes, $colNames, $length, $key, $encryptionType, $attestation)
+{
+    foreach ($dataTypes as $type) {
+
+        // Pattern matching doesn't work in AE for non-string types
+        // without an explicit cast
+        if (!dataTypeIsStringMax($type)) {
+            continue;
+        }
+
+        foreach ($patterns[$type] as $pattern) {
+
+            $patternArray = array($pattern,
+                                  $pattern."%",
+                                  "%".$pattern,
+                                  "%".$pattern."%",
+                                  );
+
+            foreach ($patternArray as $spattern) {
+
+                $unicode = dataTypeIsUnicode($type);
+                $collate = $unicode ? " COLLATE Latin1_General_BIN2" : "";
+                $PDOType = getPDOType($type);
+
+                $AEQuery = "SELECT ".$colNames[$type]."_AE FROM $tableName WHERE ".$colNames[$type]."_AE LIKE ?".$collate;
+                $nonAEQuery = "SELECT ".$colNames[$type]." FROM $tableName WHERE ".$colNames[$type]." LIKE ?".$collate;
+
+                try {
+                    $AEstmt = $conn->prepare($AEQuery);
+                    $AEstmt->bindParam(1, $spattern, $PDOType);
+                    $nonAEstmt = $conn->prepare($nonAEQuery);
+                    $nonAEstmt->bindParam(1, $spattern, $PDOType);
+                } catch (PDOException $error) {
+                    print_r($error);
+                    die("Preparing/binding statements for comparison failed");
+                }
+
+                compareResults($AEstmt, $nonAEstmt, $key, $encryptionType, $attestation, '', $type);
+            }
+        }
+    }
+}
+
+function checkErrors($errors, ...$codes)
+{
+    $codeFound = false;
+
+    foreach ($codes as $code) {
+        if ($code[0]==$errors[0] and $code[1]==$errors[1]) {
+            $codeFound = true;
+        }
+    }
+
+    if ($codeFound == false) {
+        echo "Error: ";
+        print_r($errors);
+        echo "\nExpected: ";
+        print_r($codes);
+        echo "\n";
+        die("Error code not found.\n");
+    }
+}
+
+function isEnclaveEnabled($key)
+{
+    if (strpos($key, '-enclave') !== false) return true;
+    else return false;
```

Comment:
Should always use curly braces for any if else statements, but for this one and the several cases below, you can simply do something like `return (strpos($key, '-enclave') !== false)`, etc.